storops/unity/resource/lun.py

```diff
@@ -18,14 +18,163 @@
 import logging
 
 from storops.unity.resource import UnityResource, UnityResourceList
+import storops.unity.resource.pool
+from storops.unity.enums import TieringPolicyEnum, NodeEnum, HostLUNAccessEnum
+from storops.unity.resource.storage_resource import UnityStorageResource
+from storops.exception import UnityResourceNotFoundError
 
 __author__ = 'Jay Xu'
 
 log = logging.getLogger(__name__)
 
 
 class UnityLun(UnityResource):
-    pass
+    @classmethod
+    def create(cls, cli, name, pool, size, sp=None, host_access=None,
+               is_thin=None, description=None, iolimit_policy=None,
+               is_repl_dst=None, tiering_policy=None, snap_schedule=None):
+
+        pool_clz = storops.unity.resource.pool.UnityPool
+        pool = pool_clz.get(cli, pool)
+        TieringPolicyEnum.verify(tiering_policy)
+
+        if sp is None:
+            spNode = NodeEnum.SPA
+        else:
+            spNode = sp.to_node_enum()
+            NodeEnum.verify(spNode)
+
+        if tiering_policy is None:
+            tiering_policy = TieringPolicyEnum.AUTOTIER
+
+        if description is None:
+            description = "Please specify Lun description"
+
+        # TODO:iolimit_policy adn snap_schedule
+        req_body = {
+            'name': name,
+            'description': description,
+            'replicationParameters': {
+                'isReplicationDestination': is_repl_dst
+            },
+            'lunParameters': {
+                'pool': pool,
+                'isThinEnabled': is_thin,
+                'size': size,
+                'fastVPParameters': {
+                    'tieringPolicy': tiering_policy
+                },
+                'defaultNode': spNode,
+            }
+        }
+
+        # compose host access parameters
+        if host_access is not None:
+            req_body['lunParameters']['hostAccess'] = []
+            for ha in host_access:
+                HostLUNAccessEnum.verify(ha['accessMask'])
+                req_body['lunParameters']['hostAccess'].append(
+                    {
+                        'host': ha['host'],
+                        'accessMask': ha['accessMask']
+                    }
+                )
+        # end if
+
+        resp = cli.type_action(UnityStorageResource().resource_class,
+                               'createLun',
+                               **req_body)
+        resp.raise_if_err()
+        sr = UnityStorageResource(_id=resp.resource_id, cli=cli)
+        return sr.luns[0]
+
+    def modify(self, name=None, size=None, host_access=None,
+               description=None, is_thin=None, sp=None, iolimit_policy=None,
+               is_repl_dst=None, tiering_policy=None, snap_schedule=None):
+
+        if sp is not None:

```

Comment:
Same problem here.  No need to check `None` and assign the value.  `None` value in the parameter will be filtered before sending.